stylelint-polaris/plugins/custom-properties-allowed-list/index.js

```diff
@@ -0,0 +1,181 @@
+const stylelint = require('stylelint');
+const valueParser = require('postcss-value-parser');
+
+const {
+  vendorUnprefixed,
+  matchesStringOrRegExp,
+  isCustomProperty,
+  isRegExp,
+  isString,
+} = require('../../utils');
+
+let ruleName = 'stylelint-polaris/custom-properties-allowed-list';
+
+const messages = stylelint.utils.ruleMessages(ruleName, {
+  /**
+   * @param {string} invalidProperty
+   * @param {string[]} invalidValues
+   */
+  rejected: (invalidProperty = '', invalidValues = []) => {
+    const invalidPropertyMessage = invalidProperty
+      ? `Unexpected custom property [${invalidProperty}].`
+      : null;
+
+    const invalidValuesMessage = invalidValues.length
+      ? `Invalid custom properties [${invalidValues.join(', ')}].`
+      : null;
+
+    return [invalidPropertyMessage, invalidValuesMessage]
+      .filter(Boolean)
+      .join(' ');
+  },
+});
```

Comment:
It might be nice to add some tests for these functions. This isn't blocking but might be a good follow up.

Reply:
Totally agree! There's good examples in the Stylelint built-in rules directory: https://github.com/stylelint/stylelint/tree/main/lib/rules

cc @yurm04